### PR TITLE
chore: Upgrade to `pytest-docker` v1

### LINKED ISSUE #6496: Upgrade to `pytest-docker` v1
https://pypi.org/project/pytest-docker/

`pytest-docker==1.0.0` was released earlier today, and it includes support for Docker compose v2, which is not a Python package. We'll be able to use that to drop our dev dependency on Docker compose v1, which in turn will let us upgrade to the latest major version of `jsonschema`, which is the final Meltano dependency that is considered "out of date" by `libraries.io`/`SourceRank`.

We should wait a week or two to see if any issues come up with the new major release. We're in no rush to make this change, so maybe we'll end up using `pytest-docker==1.0.1`, or something like that instead.

### PR DESCRIPTION
**Pull request by WillDaSilva**
Closes #6496

### CODE REVIEW

**Comment by WillDaSilva** [comment]
I updated the Poetry lock file using Poetry `1.1.13`, which had a bug that would set all updated entries under `[metadata.files]` to `[]`. My hope is that updating the lock file with Poetry `1.1.14`, then running the tests again should fix the issue in CI.

**Comment by edgarrmondragon** [comment]
> I updated the Poetry lock file using Poetry `1.1.13`, which had a bug that would set all updated entries under `[metadata.files]` to `[]`. My hope is that updating the lock file with Poetry `1.1.14`, then running the tests again should fix the issue in CI.

@WillDaSilva this might help https://github.com/python-poetry/poetry/issues/5988#issuecomment-1180360559

**Comment by WillDaSilva** [comment]
Keeping `flask-login` at `0.6.1` fixes [the failing tests](https://github.com/meltano/meltano/runs/7634167148).

When the tests are run in isolation, they do not error, regardless of whether I had `flask-login` at `0.6.1` or `0.6.2`, which suggests that resolving #6300 would address the underlying issue here.

Relevant changelog: https://github.com/maxcountryman/flask-login/blob/main/CHANGES.md#version-062

File: pyproject.toml [hunk 1]
```diff
@@ -38,11 +38,12 @@ fasteners = "^0.17.3"
 flask = "^2.1"
 flask-cors = "^3.0.7"
 flask-executor = "^0.10"
+flask-login = "=0.6.1" # https://github.com/meltano/meltano/issues/6548
```
Comment (edgarrmondragon) [inline pyproject.toml:41]: Is this dep syntax the same as

```suggestion
flask-login = "==0.6.1" # https://github.com/meltano/meltano/issues/6548
```

?

Comment (WillDaSilva) [reply pyproject.toml:41]: [The Poetry docs](https://python-poetry.org/docs/dependency-specification/#exact-requirements) don't have any info about using `=` or `==`. Apparently to specify an exact requirement, you're supposed to use the version number on its own. Seems like `=`, `==`, and the plain version number are all equivalent.

Comment (edgarrmondragon) [reply pyproject.toml:41]: Yeah, Poetry does its own dep parsing, though [PEP 440](https://peps.python.org/pep-0440/#version-matching) prefers `==`. Regardless, I don't have any strong preference

**Review by edgarrmondragon — APPROVED** [review]

**Comment by edgarrmondragon** [comment]
@WillDaSilva can you enable the Windows x Python 3.10 tests on this PR to see if this fixes them?

**Comment by WillDaSilva** [comment]
> @WillDaSilva can you enable the Windows x Python 3.10 tests on this PR to see if this fixes them?

I was planning to do that after this PR goes through, to avoid introducing errors to this one if something goes wrong.

**Comment by edgarrmondragon** [comment]
> I was planning to do that after this PR goes through, to avoid introducing errors to this one if something goes wrong.

@WillDaSilva sounds good 👍

**Pull request merged**